src/main/kotlin/com/mineinabyss/looty/ResourcepackGeneration.kt

```diff
@@ -0,0 +1,182 @@
+import org.bukkit.inventory.meta.LeatherArmorMeta
+import org.bukkit.inventory.meta.PotionMeta
+import java.nio.charset.Charset
+
+class ResourcepackGeneration {
+    private val tools = setOf(MaterialTags.PICKAXES, MaterialTags.AXES, MaterialTags.SHOVELS, MaterialTags.HOES, MaterialTags.SWORDS)
+    private val shieldDisplay = "{\"thirdperson_righthand\":{\"rotation\":[0,90,0],\"translation\":[10,6,-4],\"scale\":[1,1,1]},\"thirdperson_lefthand\":{\"rotation\":[0,90,0],\"translation\":[10,6,12],\"scale\":[1,1,1]},\"firstperson_righthand\":{\"rotation\":[0,180,5],\"translation\":[-10,2,-10],\"scale\":[1.25,1.25,1.25]},\"firstperson_lefthand\":{\"rotation\":[0,180,5],\"translation\":[10,0,-10],\"scale\":[1.25,1.25,1.25]},\"gui\":{\"rotation\":[15,-25,-5],\"translation\":[2,3,0],\"scale\":[0.65,0.65,0.65]},\"fixed\":{\"rotation\":[0,180,0],\"translation\":[-2,4,-5],\"scale\":[0.5,0.5,0.5]},\"ground\":{\"rotation\":[0,0,0],\"translation\":[4,4,2],\"scale\":[0.25,0.25,0.25]}}"
+    private val bowDisplay = "{\"thirdperson_righthand\":{\"rotation\":[-80,260,-40],\"translation\":[-1,-2,2.5],\"scale\":[0.9,0.9,0.9]},\"thirdperson_lefthand\":{\"rotation\":[-80,-280,40],\"translation\":[-1,-2,2.5],\"scale\":[0.9,0.9,0.9]},\"firstperson_righthand\":{\"rotation\":[0,-90,25],\"translation\":[1.13,3.2,1.13],\"scale\":[0.68,0.68,0.68]},\"firstperson_lefthand\":{\"rotation\":[0,90,-25],\"translation\":[1.13,3.2,1.13],\"scale\":[0.68,0.68,0.68]}}"
+    private val crossbowDisplay = "{\"thirdperson_righthand\":{\"rotation\":[-90,0,-60],\"translation\":[2,0.1,-3],\"scale\":[0.9,0.9,0.9]},\"thirdperson_lefthand\":{\"rotation\":[-90,0,30],\"translation\":[2,0.1,-3],\"scale\":[0.9,0.9,0.9]},\"firstperson_righthand\":{\"rotation\":[-90,0,-55],\"translation\":[1.13,3.2,1.13],\"scale\":[0.68,0.68,0.68]},\"firstperson_lefthand\":{\"rotation\":[-90,0,35],\"translation\":[1.13,3.2,1.13],\"scale\":[0.68,0.68,0.68]}}"
+
+    fun generateDefaultAssets() {
+        LootyTypeQuery.mapNotNull { it.type.item.type }.forEach { material ->
+            val root = "${looty.dataFolder.absolutePath}/assets/minecraft/models/item"
+            root.toPath().toFile().mkdirs()
+            //TODO True in most but not all cases
+            val materialFile = "$root/${material.toString().lowercase()}.json".toPath().toFile()
+            val lootyItems = LootyTypeQuery.filter { it.type.item.type == material }.map { it.type.item }
+            materialFile.createNewFile()
+            materialFile.writeText(material.generatePredicates(lootyItems), Charset.defaultCharset())
+        }
+    }
+
+    private fun Material.generatePredicates(lootyItems: List<SerializableItemStack>): String {
+        val json = JsonObject()
+        json.addProperty("parent", this.getParentModel())
+        json.add("textures", this.getTextures())
+        json.add("overrides", this.getOverrides(lootyItems))
+
+        when (this) {
+            Material.BOW -> json.add("display", bowDisplay.toJsonObject())
```

Comment:
And since you're only using values like `shieldDisplay` here, if you go this method I would just put those variables into those classes too instead of making one big one like i suggested at the top. Means we can separate concerns :)